### DISCUSSION TITLE
ESLint in subdirecotry (monorepo)

### DISCUSSION THREAD

**Discussion by ivan-mihalic** [zed-industries/zed#11330] [Feature Requests]
Hi guys,
I am facing issues with not working eslint warnings or fixing when I have opened project with nested folders (monorepo) where are eslint configs (for example frontend folder). 
when I open directly frontend folder it works. 
any advice?
thx

```
project/
 |--- frontend/
         |--- eslint.config.js
         |--- src/
 |--- backend/
         |--- eslint.config.js
         |--- src/
```

**Comment by leognmotta** [comment]
> Hi guys,
> 
> I am facing issues with not working eslint warnings or fixing when I have opened project with nested folders (monorepo) where are eslint configs (for example frontend folder). 
> 
> when I open directly frontend folder it works. 
> 
> any advice?
> 
> thx
> 
> 
> 
> ```
> 
> project/
> 
>  |--- frontend/
> 
>          |--- eslint.config.js
> 
>          |--- src/
> 
>  |--- backend/
> 
>          |--- eslint.config.js
> 
>          |--- src/
> 
> ```

This is exactly the blocker I have from using zed, can't make it work

**Reply by ivan-mihalic** [reply]
same here :(

**Comment by KaaPex** [comment]
Based on code it works only for simple projects without workspace
https://github.com/zed-industries/zed/blob/f3ddd182016cfe7e62290766678a2f42389c35c5/crates/languages/src/typescript.rs#L360

And the workspace_root is not the relative root folder:
https://github.com/zed-industries/zed/blob/f3ddd182016cfe7e62290766678a2f42389c35c5/crates/languages/src/typescript.rs#L318

**Comment by devdumpling** [comment]
+1 -- at this point now that (I think?) most of the other parts of the frontend toolchain are workable, full monorepo support (I use pnpm workspaces but doesn't really matter) is the only thing preventing me from using this as my daily driver!

I keep checking back every few months to watch progress, exciting stuff.

**Comment by snorreks** [comment]
Not wanna be that guy that just says "Any update on this?" but since it has been one year.... Any update on this?

**Comment by radiantbeing** [comment]
I hope this helps - I encountered a similar issue and wanted to share what worked for me, though your mileage may vary.

**Potential solution:** You might want to try enabling `useFlatConfig` in your Zed settings:

`.zed/settings.json`:
```json
{
   "lsp": {
       "eslint": {
           "settings": {
               "experimental": {
                   "useFlatConfig": true
               }
           }
       }
   }
}
```

Example that worked for me: https://stackblitz.com/edit/stackblitz-starters-bp4xezmq?file=.zed%2Fsettings.json&view=editor

If you'd like to test it, you can download(cloud icon above file explorer) the example and run `npm install` to see if it works in your environment as well.

It has a single `package.json` at the project root, with separate `client` and `server` folders each containing their own `eslint.config.js`. After applying this setting, ESLint rules in each directory seemed to work independently for me.

`client/index.js`:

<img width="1373" alt="스크린샷 2025-06-27 오후 1 17 33" src="https://github.com/user-attachments/assets/e2848618-75c5-46fe-b68b-551102fd4390" />

`server/index.js`:

<img width="1373" alt="스크린샷 2025-06-27 오후 1 17 41" src="https://github.com/user-attachments/assets/0aa7410c-368d-4498-8601-bc3222c6f7ef" />


I hope this might be helpful for your case, but please let me know if you need any clarification or if this doesn't work for your setup.

**Reply by LeNei** [reply]
This worked for me :D

**Reply by TQTuyen** [reply]
Thanks, it worked for me

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
